### PR TITLE
Disable add-on name unique checks for new submissions.

### LINKED ISSUE #3094: Remove name uniqueness checks for new add-on submissions
There's a bug with name resolution for localized names in WebExtensions currently.

We decided that we remove uniqueness checks for new add-on submissions (no matter if listed or unlisted) but keep the check for editing existing add-ons since they don't go through review again.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #3094

### CODE REVIEW

**Comment by andymckay** [comment]
Does this occur in the API as well, or does it do no checks because its unlisted?

**Comment by EnTeQuAk** [comment]
Correct, the API only recently gained the feature to create an Add-on (and only for WebExtensions) and only create unlisted add-ons which were never checked for unique names.

File: src/olympia/devhub/forms.py [hunk 1]
```diff
@@ -551,10 +551,8 @@ class NewAddonForm(AddonUploadForm):
     def clean(self):
         if not self.errors:
             self._clean_upload()
-            xpi = parse_addon(self.cleaned_data['upload'])
-            # We don't enforce name uniqueness for unlisted add-ons.
-            if not self.cleaned_data.get('is_unlisted', False):
-                clean_addon_name(xpi['name'], addon_type=xpi['type'])
```
Comment (eviljeff) [inline src/olympia/devhub/forms.py]: what does `clean_addon_name` do - I guess the actual name uniqueness checking is in there making it redundant now?

Comment (EnTeQuAk) [inline src/olympia/devhub/forms.py]: Actually, `clean_addon_name` only checks for name collisions but we still need that to check for collisions if someone edits an add-on.

**Comment by tofumatt** [comment]
r+ from me, assuming the check exists elsewhere for existing add-ons like we want (which I assume it does, as you only changes the tests for creating an add-on).

File: src/olympia/devhub/forms.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/devhub/forms.py]: I'm not sure I follow the logic that we would still check for collisions for existing add-ons but not new, but 😕 
@jvillalobos

Comment (EnTeQuAk) [inline src/olympia/devhub/forms.py]: The problem with existing ones compared to new ones is that they don't go through review again once you change any detail like the add-on name.

Comment (jvillalobos) [inline src/olympia/devhub/forms.py]: I noted on IRC that I don't think we should keep the check on edit either. We have abuse reports and other ways to learn if an add-on is doing something sketchy.

**Pull request merged**